### PR TITLE
Usage of new `EthClient`

### PR DESCRIPTION
**Pull request by goran-ethernal**
# Description

This PR replaces the `ethgo.Client` with the new implementation added in `blade`: https://github.com/Ethernal-Tech/blade/pull/160 throughout the code.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

**Review by dusannosovic-ethernal — DISMISSED** [review]

File: types/state_tx.go [hunk 1]
```diff
@@ -200,12 +200,25 @@ func (tx *StateTx) copy() TxData { //nolint:dupl
 	return cpy
 }
 
+func (tx *StateTx) marshalJSON(a *fastjson.Arena) *fastjson.Value {
+	v := a.NewObject()
+
+	tx.BaseTx.marshalJSON(a, v)
+	v.Set("type", a.NewString(fmt.Sprintf("0x%x", tx.transactionType())))
```
Comment (Stefan-Ethernal) [inline types/state_tx.go:207]: Why not moving the `type` marshaling into the `BaseTx`?

File: types/json.go
Comment (Stefan-Ethernal) [inline types/json.go:1]: Rename the file to `json_marshalling_util.go`. WDYT?

File: e2e-polybft/e2e/jsonrpc_test.go [hunk 2]
```diff
@@ -25,447 +22,6 @@ var (
...
-		txReceipt := txn.Receipt()
-
-		// Use a wrapper function from "jsonrpc" package when the config is introduced.
-		trace, err := debug.TraceTransaction(txReceipt.TransactionHash, jsonrpc.TraceTransactionOptions{})
-		require.NoError(t, err)
-		require.Equal(t, txReceipt.GasUsed, trace.Gas)
-	})
-}
-
-func TestE2E_JsonRPC_NewEthClient(t *testing.T) {
```
Comment (Stefan-Ethernal) [inline e2e-polybft/e2e/jsonrpc_test.go:468]: Rename it now to `TestE2E_JSONRPC`?

File: e2e/txpool_test.go [hunk 3]
```diff
@@ -294,39 +294,31 @@ func TestTxPool_RecoverableError(t *testing.T) {
...
-	secondTx, err := client.Eth().GetTransactionByHash(hashes[1])
+	secondTx, err := client.GetTransactionByHash(hashes[1])
 	require.NoError(t, err)
 	require.NotNil(t, secondTx)
-
-	// first two are in one block
-	require.Equal(t, firstTx.BlockNumber, secondTx.BlockNumber)
-
-	// last tx is included in next block
-	require.NotEqual(t, secondTx.BlockNumber, receipt.BlockNumber)
```
Comment (Stefan-Ethernal) [inline e2e/txpool_test.go]: What happened with these assertions?

**Review by Stefan-Ethernal — DISMISSED** [review]

File: types/state_tx.go (hunk 1 above)
Comment (goran-ethernal) [reply types/state_tx.go:207]: `BaseTx` doesn't know the type. `Type` is not a field, just a function on specific transaction types, which is fine, there is no need for a setter.

File: e2e/txpool_test.go (hunk 3 above)
Comment (goran-ethernal) [reply e2e/txpool_test.go]: I will revert these. Removed it on the first iteration, and forgot to revert them back.

File: e2e-polybft/e2e/jsonrpc_test.go (hunk 2 above)
Comment (goran-ethernal) [reply e2e-polybft/e2e/jsonrpc_test.go:468]: Fix: https://github.com/Ethernal-Tech/blade/pull/165/commits/6a43df12d00f176d42daa16a06305926806ccfcd

File: e2e/txpool_test.go (hunk 3 above)
Comment (goran-ethernal) [reply e2e/txpool_test.go]: Fix: https://github.com/Ethernal-Tech/blade/pull/165/commits/6a43df12d00f176d42daa16a06305926806ccfcd

**Review by Stefan-Ethernal — APPROVED** [review]

**Pull request merged**